thunder/tests/opinfos.py

```diff
@@ -4992,6 +4993,52 @@ def amax_amin_sample_generator(op, device, dtype, requires_grad, **kwargs):
 reduction_ops.append(amin_opinfo)
 
 
+def max_sample_generator(op, device, dtype, requires_grad, **kwargs):
+    # For grad test stability it's better to use wider range of values
+    make = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad, low=-1000, high=1000)
+
+    # shape, dim, keepdim
+    cases = (
+        ((2, 2, 3), 1, True),
+        ((2, 3, 1), 0, False),
+    )
```

Comment:
How does it work with NaNs, infs?

Reply:
It always returns the first occurence.

```python
import torch
import thunder

def foo(*args):
    return torch.max(*args)

jfoo = thunder.jit(foo)

for dev in ("cpu", "cuda"):
    x1 = torch.tensor([float("nan"), float("nan")])
    x2 = torch.tensor([float("inf"), float("inf")])

    def _test(*args):
        expected = foo(*args)
        actual = jfoo(*args)
        torch.testing.assert_close(actual, expected, equal_nan=True)

    for x in (x1, x2):
        _test(x)
        # Test that same indices are returned
        _test(x, 0)
        # Test that same indices are returned
        _test(x, 0, True)
        _test(x, x)

```

However, to add test with multiple repeating values - we will have to add a fix for this problem from the PR desc

> Problem: However, using amax leads to a difference in gradient computation.

> amax/amin evenly distributes gradient between equal values, while max(dim)/in(dim) propagates gradient only to a single index in the source tensor.

Reply:
Let's definitely expand the comment to describe what happens in this case!

Reply:
Follow-up seems like a good plan. Thank you.